src/uct/tcp/tcp_ep.c

```diff
@@ -571,6 +571,37 @@ static void uct_tcp_ep_handle_disconnected(uct_tcp_ep_t *ep, ucs_status_t status
     }
 }
 
+static inline ucs_status_t uct_tcp_ep_handle_send_err(uct_tcp_ep_t *ep,
+                                                      ucs_status_t status)
+{
+    uct_tcp_iface_t *iface = ucs_derived_of(ep->super.super.iface,
+                                            uct_tcp_iface_t);
+
+    ucs_assert(status != UCS_ERR_NO_PROGRESS);
+
+    status = uct_tcp_ep_handle_io_err(ep, "send", status);
+    if (status == UCS_ERR_CANCELED) {
+        /* If no data were read to the allocated buffer,
+         * we can safely reset it for futher re-use and to
+         * avoid overwriting this buffer, because `rx::length == 0` */
+        if (ep->tx.length == 0) {
+            uct_tcp_ep_ctx_reset(&ep->tx);
+        }
+    } else {
+        uct_tcp_ep_handle_disconnected(ep, status);
+        if (iface->super.err_handler != NULL) {
+            /* Translate the error to EP timeout error, since user expects that
+             * status returned from UCT error callback is the same as it is
+             * returned from the UCT EP send operation (or from UCT competion
+             * callback)
+             * TODO: revise this behavior */
+            return UCS_ERR_ENDPOINT_TIMEOUT;
```

Comment:
maybe uct_tcp_ep_set_failed should pass the status instead of always passing UCS_ERR_ENDPOINT_TIMEOUT?

Reply:
unfortunately, our gtests expect `UCS_ERR_ENDPOINT_TIMEOUT` and then replaced function pointers return `UCS_ERR_ENDPOINT_TIMEOUT`

Reply:
ok